tests/functional/instance_manager/__init__.py

```diff
@@ -0,0 +1,122 @@
+# coding=utf-8
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+"""Setup for the package tests."""
+
+import copy
+import json
+import multiprocessing
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import time
+import uuid
+from subprocess import CalledProcessError
+
+from libtest import get_free_port
+from libtest import project
+from libtest import codechecker
+from libtest import env
+
+# Stopping events for CodeChecker servers.
+__SERVER_EVENTS = [multiprocessing.Event(),
+                   multiprocessing.Event(),
+                   multiprocessing.Event()]
+
+# Test workspace initialized at setup for authentication tests.
+TEST_WORKSPACE = None
+
+
+def setup_package():
+    """Setup the environment for the tests then start the server."""
+
+    global TEST_WORKSPACE
+    TEST_WORKSPACE = env.get_workspace('instances')
+
+    os.environ['TEST_WORKSPACE'] = TEST_WORKSPACE
+
+    test_config = {}
+
+    # Setup environment varaibled for the test cases.
+    host_port_cfg = env.get_host_port_cfg()
+
+    codechecker_cfg = {
+        'use_postgresql': False,
+        'workspace': TEST_WORKSPACE,
+        'run_names': [],
+        'checkers': []
+    }
+
+    codechecker_cfg.update(host_port_cfg)
+
+    test_config['codechecker_1'] = codechecker_cfg
+
+    # We need a second server
+    codechecker_cfg = {
+        'use_postgresql': False,
+        'workspace': TEST_WORKSPACE,
+        'run_names': [],
+        'checkers': []
+    }
+    host_port_cfg = env.get_host_port_cfg()
+    if host_port_cfg['viewer_port'] == \
+            test_config['codechecker_1']['viewer_port']:
+        host_port_cfg['viewer_port'] = int(host_port_cfg['viewer_port']) + 1
+    codechecker_cfg.update(host_port_cfg)
+    test_config['codechecker_2'] = codechecker_cfg
+
+    # Export configuration for the tests.
+    env.export_test_cfg(TEST_WORKSPACE, test_config)
+
+    # Wait for previous test instances to terminate properly and
+    # clean the instance file in the user's home directory.
+    time.sleep(5)
+
+
+def teardown_package():
+    """Stop the CodeChecker server."""
+
+    for ev in __SERVER_EVENTS:
+        ev.set()
+
+    # TODO If environment variable is set keep the workspace
+    # and print out the path.
+    global TEST_WORKSPACE
+
+    print("Removing: " + TEST_WORKSPACE)
+    shutil.rmtree(TEST_WORKSPACE)
+
+
+def set_event(stop_event_idx):
```

Comment:
Maybe a `set_stop_event` name will be a bit more descriptive about what it does.